### ISSUE TITLE
Handle (ignore?) duplicate SetBitmapAttrs()

### ISSUE THREAD

**Issue by travisturner** [pilosa/pilosa#450]
### Description
For every SetBit on a categorized Bitmap, Umbel sends two queries:
```
SetBit(row=1, col=200, frame="f")
SetBitmapAttrs(row=1, frame="f", category=456)
```
They do this because for them it's easier to just set category every time than to track the first SetBit on a Bitmap, especially since they are often writing to multiple DBs for every profile (they maintain a tree of DBs representing their property hierarchy).

Currently in Pilosa, when we receive a flood of duplicate SetBitmapAttrs() to the same bitmap, there is a fairly significant overhead in writing that attribute every time. Since most of the time nothing is changing, the proposal is to put some sort of cache/check layer that will prevent writes to BitmapAttrs when nothing has changed.

@tgruben suggested keeping an in-memory bloom filter that is checked on SetBitmapAttrs(). If the key is found in the bloom filter, then no-op. Because it's possible that the SetBitmapAttrs() could be setting an Attr that already exists but the value is changing, the key checked against the bloom filter needs to include the value (although this could be a problem if the user wanted to change Attr value back to a previous value... in that case cache-invalidation becomes a concern):
```
BitmapAttr: db, frame, row-id, attr-name, attr-value
ProfileAttr: db, col-id, attr-name, attr-value
``` 
Question: should we also handle this on SetProfileAttrs()?

Note: This request came from Umbel, and is specific to their use case. With that said, it's possible that other users might send data to Pilosa in the same way.

See @tgruben for a sample reproduction of the problem.

### Success criteria (What criteria will consider this ticket closeable?)
One can flood Pilosa with duplicate SetBitmapAttrs() calls with little degradation to performance.

**Comment by travisturner** [comment]
@tgruben is going to setup a simple benchmark for this scenario and coordinate with @benbjohnson  on implementing his suggestion and then comparing benchmark results.

**Comment by travisturner** [comment]
fixed by #562
